### DISCUSSION TITLE
ppa update vs re-image

### DISCUSSION THREAD

**Discussion by davidstoll** [Joshua-Riek/ubuntu-rockchip#312] [General]
I apologize if this has been asked and answered, but I was away for a bit and coming back to this awesome project.

Is it possible to just apt update/upgrade or is it still best to delete and reimage?  If we can update/upgrade, at what version can we start doing this and is there anything special we have to setup to move forward that way?  Also, I forgot what version I left off on (probably in the high teens 1.1x, but I don't know for sure), so is if you could help me know what version I have, that would be helpful too.  If it matters, I use the Orange PI 5 and 5b.

Thank you again, this is an awesome project!

**Comment by Joshua-Riek** [comment]
What is the output of `uname -a` I might be able to tell from kernel version? I actually don't have anything that says the exact version info, i should probably do that, was not expecting this project to grow this much. 

Any changes to the kernel, bootloader, and firmware will be able to download from a simple apt update and upgrade. These three are what will have the most changes and new features. However when it comes to other changes they will not be able to install with an apt update. An example would be if I change what packages are pre-installed, or create a new systemd service to start something when the operating system starts. 

There is an example of this and more information in #308, where i needed to add a startup service to fix audio for the ROCK 5. I could not distribute this change to previously installed systems.

**Comment by davidstoll** [comment]
Thanks, that makes sense.

Linux opi5b 5.10.160-rockchip #5 SMP Wed Jun 28 00:44:29 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux
